src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -328,6 +338,11 @@ public void jobLogic () {
             // Set baseUrl after success.
             status.baseUrl = otpServer.publicUrl;
         }
+
+        // A new FetchSingleFeedJob could be started after the AutoDeployJob has made sure no fetching jobs exist and
+        // the DeployJob has started. Therefore this new feed version wouldn't result in a new DeployJob getting kicked
+        // off since there was already one running. To catch this new feed version another auto deploy job is started.
+        addNextJob(new AutoDeployJob(deployment.parentProject(), owner));
```

Comment:
This won't be executed in a number of code paths. Particularly for deploying to EC2, the `replaceEC2Servers()` method will be ran and then this method will have a return called before this line is reached (see line 326). There are quite a few places in this method with a return statement, so I'm not sure how best to make sure this AutoDeployJob is always added, or if it should always be added. Might be worth having a technical call about charting out under which circumstances auto-deployment should be disabled or if should always be ran again and again even if it is error-ing out each time.

Reply:
Maybe this could just be moved to the `DeployJob#jobFinished` method

Reply:
Hey @br648, sorry for taking so long to reply and look into this. I think Landon's suggestion will probably work. And I think it should be placed at the very end of the `jobFinished` method and also it would probably make sense to only activate this code if the `status.error` variable is false.

Reply:
No worries! Updated.